optuna/storages/rdb/storage.py

```diff
@@ -713,7 +726,7 @@ def _get_and_cache_trial(self, trial_id, deepcopy=True):
         # Terminate transaction explicitly to avoid connection timeout during transaction.
         self._commit(session)
 
-        return frozen_trial
+        return copy.deepcopy(frozen_trial) if deepcopy else frozen_trial
```

Comment:
This seems a bug introduced by the commit https://github.com/optuna/optuna/commit/0325b95874595ddbd942d930a35ad26e295fdbf1#diff-70f8b9dc0cc3d59bab2c32029cf14808. Thank you for fixing it!